docs/admin/reference-architectures.md

```diff
@@ -0,0 +1,163 @@
+handle dashboard connections or API calls.
+
+### Provisioner
+
+Provisioners in Coder execute Terraform during workspace and template builds.
+While the platform includes built-in provisioner daemons by default, there are
+advantages to employing external provisioners. These external daemons provide
+secure build environments, and reduce server load, improving performance and
+scalability. Each provisioner can handle a single concurrent workspace build,
+allowing for efficient resource allocation and workload management.
+
+### Registry
+
+The Coder Registry hosts starter templates for various cloud providers and
+orchestration platforms, enabling self-service cloud development environments
+via Terraform-defined infrastructure. Additionally, Coder introduces _Modules_
+to streamline template creation by extracting commonly used functionalities such
+as web IDEs, third-party integrations, and helper scripts into reusable
+components.
+
+The Registry is hosted service and it is not available for air-gapped
+deployments.
+
+## Scale tests methodology
+
+Scaling Coder involves careful planning and testing to ensure it can handle more
+load without compromising service. This process encompasses infrastructure
+setup, traffic projections, and aggressive testing to identify and mitigate
+potential bottlenecks.
+
+A dedicated cluster for Coder is Kubernetes cluster specifically configured to
+host and manage Coder workloads. Kubernetes provides container orchestration
+capabilities, allowing Coder to efficiently deploy, scale, and manage workspaces
+across a distributed infrastructure. This ensures high availability, fault
+tolerance, and scalability for Coder deployments. The cluster can be deployed
+using the Helm chart.
+
+Our scale tests include the following stages:
+
+1. Prepare environment: create expected users and provision workspaces.
+
+2. Dashboard evaluation: verify the responsiveness and stability of Coder
+   dashboards under varying load conditions. This is achieved by simulating user
+   interactions using instances of headless Chromium browsers.
+
+3. SSH connections: establish user connections with agents, verifying their
+   ability to echo back received content.
+
+4. Workspace application traffic: assess the handling of user connections with
+   specific workspace apps, confirming their capability to echo back received
+   content effectively.
+
+5. Cleanup: delete workspaces and users created in step 1.
+
+### Infrastructure and setup requirements
+
+In a single workflow, the scale tests runner evenly spreads out the workload
+like this:
+
+- 80% of users open and utilize SSH connections.
+- 25% of users connect to the workspace using the Web Terminal.
+- 40% of users simulate traffic for workspace apps.
+- 20% of users interact with the Coder UI via a headless browser.
+
+This distribution closely mirrors natural user behavior, as observed among our
+customers.
```

Comment:
A big reason for the high SSH rating is that vs code and jetbrains are used over SSH. Maybe we can point this out as well?

Reply:
Done 👍